scripts/pi-hole/js/gravity.js

```diff
@@ -47,9 +47,18 @@ function eventsource() {
               // Enqueue the next data chunk into our target stream
               controller.enqueue(value);
               var string = new TextDecoder().decode(value);
-              // Remove ${OVER} from the string
-              string = string.replaceAll("\r[K", "\n");
-              ta.append(string);
+
+              // If a Carriage Return is found ...
+              if (string.indexOf("\r") === -1) {
+                ta.append(string);
+              } else {
+                // ... remove the last line from the output ...
+                ta.text(ta.text().substring(0, ta.text().lastIndexOf("\n")) + "\n");
+                // ... and append the new text to the end of the output,
+                // without ${OVER} ("CR + ESC[K") or Carriage Return.
+                ta.append(string.replaceAll("\r[K", "").replaceAll("\r", ""));
```

Comment:
This `.replaceAll()` kinda feels wrong to me but being aware of the one-line-after-another chunking, it is still doing what it should - even when it doesn't feel right (subjective feeling all the way).

Reply:
We can create a function to proper cleanup the string, removing "\r[K", "\r". 
We can even allow the use of color codes and replace them with HTML `<span>` tags with CSS classes to use color output (like we had in v5).

Reply:
Yeah, color codes may be even nicer but it may be over the top. If we want to go entirely crazy we could even make FTL send `<span ...> ... </span>`  already by itself when requesting via the API.

Reply:
Let's not get crazy here... formatting should be done by the receiving/displaying component, not the sending one, I think.